src/Interface/ResultTypeInferenceInterface.cpp

```diff
@@ -0,0 +1,20 @@
+//===------------ ResultTypeInferenceInterface.cpp --------------===//
+//===--------- Infer Data Type for Results Interface Definition --------===//
+//
+// Copyright 2020 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains the declaration of the data type reference  interface.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Interface/ResultTypeInferenceInterface.hpp"
```

Comment:
It would be more consistent to drop `src/Interface/` because this is the file that implements the declarations specified in  `ResultTypeInferenceInterface.hp`.